homeassistant/components/weenect/__init__.py

```diff
@@ -0,0 +1,151 @@
+"""The Weenect integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+import re
+from typing import Any
+
+from _collections_abc import Callable
+from aioweenect import AioWeenect
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, TRACKER_ADDED, TRACKER_REMOVED
+
+PLATFORMS = [
+    Platform.DEVICE_TRACKER,
+]
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+DEFAULT_UPDATE_RATE = 30
+
+DURATION_PATTERN = re.compile(r"\d\d[S,M,H]")
```

Comment:
This implies the duration always has two digits, which sounds like an assumption that could break at any point.

Additionally, this will happily match e.g. `"६೬,"` because of two bugs:
* `\d` matches any decimal digit, not just the ASCII 0-9, by default. You can add the `re.ASCII` flag, or explicitly specify `[0-9]`. (The characters above are Devanagari and Kannada decimal digits.)
* You don't need commas in a character class.

I'd go with 
```suggestion
DURATION_PATTERN = re.compile(r"^[0-9]+[SMH]$")
```
and then maybe upper-case the duration string in `parse_duration`.

Reply:
As an aside, maybe this transformation (and the function) should be in `aioweenect`? It sounds like it's related to the remote data format.

Reply:
Thanks for the feedback. Currently aioweenect returns data from the API "as-is".
But does make sense for me to turn the response to pythonic data structures.